### ISSUE TITLE
Agentic coding for automatic kernel generation and **optimization**

### ISSUE THREAD

**Issue by learning-chip** [huawei-csl/pto-kernels#24]
## Overall plan

Four steps, from low-hanging fruit to more advanced:
1. Extract performance-tuning knowledge from existing AscendC/CCE/PTO open-source repos and doc (everything in https://gitcode.com/cann), summarize into skills (basically step-by-step tutorials)
  -> support cursor vide coding (50% automation)
2. Use online LLM API + local docker-ascend sandbox, for autonomous kernel dev &  testing cycle 
   -> runs overnight without frequent human interactions (100% automation，50% local)
3. Switch to local LLM deployment (vllm-ascend/sglang) to host open-source coding models
   ->100 % local, reduce token bills, and enable step 4
4. SFT & RL on open-weight models, absorb NPU kernel acknowledge from context into weights

Prioritize steps 1~2

## Execution plan

A first easy step is to let agent generate all the [nonlinear activation functions](https://docs.pytorch.org/docs/stable/nn.functional.html#non-linear-activation-functions).
- Input (in addition to skills): Pytorch reference code
- Output: PTO-ISA kernel code (forward + backward), build script (cmake + torch binding), unit test, perf measurement

Those are all simple vector kernels, only need to teach agent: use long-enough SIMD length, use double-buffer, mask-out tail blocks, how to calculate and compare to expected peak performance (roofline)

Then move to more complex Cube and Mix kernels.
Most importantly, we can compare agent-generated software pipelining (set/wait flags) vs compiler-generated :D

## References

- FlashInfer-Bench https://arxiv.org/abs/2601.00227
- For Ascend NPU: https://arxiv.org/abs/2601.00227, https://arxiv.org/abs/2601.22760
- https://github.com/huggingface/kernels/tree/main/skills/cuda-kernels
- https://github.com/flashinfer-ai/flashinfer/tree/main/.claude/skills
